### PR TITLE
[Windows] Refactor virtual-env image generation provisioners

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are improving and refactoring bunch of virtual-environments provisioners.

#### Related issue:
https://github.com/actions/virtual-environments-internal/issues/422

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/ImageHelpers/ImageHelpers.psm1 [hunk 1]
```diff
@@ -5,6 +5,7 @@ param()
 . $PSScriptRoot\InstallHelpers.ps1
 . $PSScriptRoot\ChocoHelpers.ps1
 . $PSScriptRoot\TestsHelpers.ps1
+. $PSScriptRoot\InstallVSHelpers.ps1
```
Comment (maxim-lobanov) [inline images/win/scripts/ImageHelpers/ImageHelpers.psm1:8]: `VisualStudioHelpers`

File: images/win/scripts/Installers/Initialize-VM.ps1 [hunk 2]
```diff
@@ -97,7 +112,9 @@ else {
 }
 
 # Run the installer
-[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor "Tls12"
+if (Test-IsWin19) {
+    [Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor "Tls12"
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Initialize-VM.ps1:116]: Let's merge with condition above

File: images/win/scripts/Installers/Initialize-VM.ps1 [hunk 3]
```diff
@@ -67,7 +77,12 @@ Write-Host "Disable IE ESC"
 Disable-InternetExplorerESC
 
 Write-Host "Setting local execution policy"
-Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope LocalMachine  -ErrorAction Continue | Out-Null
+if (Test-IsWin16) {
+    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope MachinePolicy  -ErrorAction Continue | Out-Null
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Initialize-VM.ps1:81]: Let's set the single security policy

File: images/win/scripts/ImageHelpers/InstallVSHelpers.ps1 [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+function Get-VisualStudioPath {
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [string]$Version,
+        [Parameter(Mandatory=$true)]
+        [string]$Edition
+    )
+
+    return "${env:ProgramFiles(x86)}\Microsoft Visual Studio\${Version}\${Edition}"
```
Comment (maxim-lobanov) [inline images/win/scripts/ImageHelpers/InstallVSHelpers.ps1:165]: Let's use Get-VSSetupInstance

File: images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1 [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        The URL from which the bootstrapper will be downloaded. Required parameter.
+
+    .PARAMETER WorkLoads
+        The string that contain workloads that will be passed to the installer.
+    #>
+
+    Param
+    (
+        [Parameter(Mandatory)]
+        [String] $BootstrapperUrl,
```
Comment (andy-mishechkin) [inline images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1:20]: Nitpick: perhaps will looks more readable as this:
 [Parameter(Mandatory)] [String] $BootstrapperUrl,
 [Parameter()] [String] $WorkLoads

Comment (dibir-magomedsaygitov) [reply images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1:20]: We are using this approach for every other function, so let's leave it as it is if you don't mind.

Comment (andy-mishechkin) [reply images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1:20]: ok

**Pull request merged**